From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 05 Oct 2015 20:39:56 +0100
Subject: block: Avoid ABI change in 4.2.2
Forwarded: not-needed

struct blk_flush_queue is only used in the block core; its definition
is not available to OOT modules nor used by in-tree modules.  Hide the
change from genksyms.

Also hide a new #include from genksyms.

---
--- a/block/blk.h
+++ b/block/blk.h
@@ -23,12 +23,15 @@ struct blk_flush_queue {
 	struct list_head	flush_data_in_flight;
 	struct request		*flush_rq;
 
+	spinlock_t		mq_flush_lock;
+
+#ifndef __GENKSYMS__
 	/*
 	 * flush_rq shares tag with this rq, both can't be active
 	 * at the same time
 	 */
 	struct request		*orig_rq;
-	spinlock_t		mq_flush_lock;
+#endif
 };
 
 extern struct kmem_cache *blk_requestq_cachep;
--- a/block/blk-flush.c
+++ b/block/blk-flush.c
@@ -73,7 +73,9 @@
 
 #include "blk.h"
 #include "blk-mq.h"
+#ifndef __GENKSYMS__
 #include "blk-mq-tag.h"
+#endif
 
 /* FLUSH/FUA sequences */
 enum {
